src/shredder.rs

```diff
@@ -229,6 +244,21 @@ impl Shredder for RegularShredder {
     fn deshred(
         shreds: &[Option<ValidatedShred>; TOTAL_SHREDS],
     ) -> Result<(Slice, [ValidatedShred; TOTAL_SHREDS]), DeshredError> {
+        let () = validate_shred_indexes(&shreds)?;
+        for i in 0..DATA_SHREDS {
```

Comment:
could avoid a lot of duplication here by turning this into a function `validate_shred_layout` that takes something like `num_data` and `num_coding` and returns `Result<(), DeshredError>`

Reply:
could also subsume `validate_shred_indexes`

Reply:
then validate_shred_layout also does not need to take them as arguments, can just be a provided method of the `Shredder` trait

Reply:
good idea, now done.